lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp

```diff
@@ -367,14 +392,45 @@ void mlir::torch::onnx_c::populateDefaultDomainQtoZ(
                                            binder.op->getRegions().size())
                 .getResult(0);
 
-        Value outScale = rewriter.create<Torch::AtenMulFloatOp>(
-            binder.getLoc(), rewriter.getType<Torch::FloatType>(), aScale,
-            bScale);
-        Value outZp = rewriter.create<Torch::ConstantIntOp>(
-            binder.getLoc(), rewriter.getType<Torch::IntType>(),
-            rewriter.getIntegerAttr(rewriter.getIntegerType(64), 0));
-        c = rewriter.create<Torch::Aten_MakePerTensorQuantizedTensorOp>(
-            binder.getLoc(), cTy, c, outScale, outZp);
+        bool aScaleIsTensor = isa<Torch::ValueTensorType>(aScale.getType());
+        bool bScaleIsTensor = isa<Torch::ValueTensorType>(bScale.getType());
+
+        if (!aScaleIsTensor && !bScaleIsTensor) {
+
+          Value outScale = rewriter.create<Torch::AtenMulFloatOp>(
+              binder.getLoc(), rewriter.getType<Torch::FloatType>(), aScale,
+              bScale);
+          Value outZp = rewriter.create<Torch::ConstantIntOp>(
+              binder.getLoc(), rewriter.getType<Torch::IntType>(),
+              rewriter.getIntegerAttr(rewriter.getIntegerType(64), 0));
+          c = rewriter.create<Torch::Aten_MakePerTensorQuantizedTensorOp>(
+              binder.getLoc(), cTy, c, outScale, outZp);
+        }
+        if (aScaleIsTensor || bScaleIsTensor) {
```

Comment:
I know semantically this will not happen, but do we want to ensure both are a tensor? And I assume bScale is the bias here? So it can be None.

Reply:
I am missing a case here. In theory all permutations are valid. I am missing the case where both are tensors. 

I think a is input, b is weight and c is bias, btw.

Reply:
Thanks for catching this!